src/ui/src/app/my-team/my-team.component.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, ElementRef, ViewChild, inject } from '@angular/core';
+import { Router } from '@angular/router';
+import mermaid from 'mermaid';
+import { EmployeeServiceService } from '../Service/employee-service.service';
+
+@Component({
+  selector: 'app-my-team',
+  standalone: true,
+  imports: [],
+  templateUrl: './my-team.component.html',
+  styleUrl: './my-team.component.scss'
+})
+export class MyTeamComponent {
+
+  @ViewChild('mermaidDiv', { static: false }) mermaidDiv!: ElementRef;
+  router = inject(Router)
+  employeeData: any[] = [];
+  employeeDataModal: any[] = [];
+  isDataLoaded: boolean = false;
+  service = inject(EmployeeServiceService)
+
+  ngOnInit(): void {
+    this.loadEmployeeHierarchy();
+    (window as any).onA = (nodeName: string) => {
+      console.log('Node clicked:', nodeName);
+      this.router.navigate(['/employee-display']);
+    };
+
+    mermaid.initialize({
+      securityLevel: 'loose',
+      theme: 'base',
+      flowchart: {
+        nodeSpacing: 100,
+        rankSpacing: 80,
+        padding: 20
+      },
+
+      themeVariables: {
+        primaryColor: '#0369a1',
+        primaryTextColor: '#ffffff',
+        lineColor: '#7f8b9b',
+        nodeBorder: '#075985',
+
+      },
+      themeCSS: `
+        .node rect {
+          rx: 8px;                    /* Rounded corners */
+          height: 60px !important;    /* Fixed height for nodes */
+          transition: all 0.3s ease;
+        }
+        .node:hover rect {
+          fill: #113c60 !important;  
+        }
+        .node:hover text {
+          fill: #ffffff !important;
+        }
+      `
+
+    });
+
+  }
+  viewTeamHierarchy() {
+    this.service.getMyTeamHierarchy(true).subscribe({
+      next: (response: any) => {
+        if (response) {
+          this.employeeDataModal = response;
+          this.isDataLoaded = true;
+          this.renderMermaidChart();
+        } else {
+          alert('No Employee found');
+        }
+      },
+      error: (error) => {
+        console.error('Error Displaying Hierarchy:', error);
+        alert('An error occurred while displaying hierarchy');
+      },
+    });
+  }
+
+  loadEmployeeHierarchy() {
+    this.service.getMyTeamHierarchy(false).subscribe({
```

Comment:
variable name should be proper, service should be like it id EmployeeService then variable name is like employeeService